components/renku_data_services/project/db.py

```diff
@@ -136,25 +165,43 @@ async def update_project(self, user: base_models.APIUser, project_id: str, **pay
 
                 project = projects[0]
                 visibility_before = project.visibility
-
-                if "repositories" in payload:
-                    payload["repositories"] = [
-                        schemas.ProjectRepositoryORM(url=r, project_id=project_id, project=project)
-                        for r in payload["repositories"]
-                    ]
-
-                for key, value in payload.items():
-                    # NOTE: ``slug``, ``id``, ``created_by``, and ``creation_date`` cannot be edited
-                    if key not in ["slug", "id", "created_by", "creation_date"]:
-                        setattr(project, key, value)
-
-                if visibility_before != project.visibility:
-                    public_project = project.visibility == Visibility.public
-                    await self.project_authz.update_project_visibility(
-                        requested_by=user, project_id=project_id, public_project=public_project
-                    )
-
-                return project.dump()  # NOTE: Triggers validation before the transaction saves data
+        match project.visibility:
+            case Visibility.private | Visibility.private.value:
+                vis = MsgVisibility.PRIVATE
+            case Visibility.public | Visibility.public.value:
+                vis = MsgVisibility.PUBLIC
+            case _:
+                raise NotImplementedError(f"unknown visibility:{project.visibility}")
```

Comment:
This will send the wrong visibility! Has this change been tested through a deployment?